### PR TITLE
Adding option doctor job to become a surgeon just for RP.

### PR DESCRIPTION
**Pull request by Huffie56**
# About the pull request
i am adding an option to become Surgeon as an only RP purpose.
remove most of the gear those jobs (nurse, doctor, research)spawn with and put it in their respective vendor.

create a parent for medical preset to avoid to having three time the same thing.

# Explain why it's good for the game
The idea is to show to other people that you are a doctor specialized in surgery.

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

change in character preference here you can choose between doctor or surgeon.
image
Surgeon.
image
Doctor.
image

</details>

# Changelog

:cl:
add: Added the option for the doctor job to become a surgeon so you can choose to RP one.
add: Added the option in doctor vendor for medical belts, medical set, surgical caps, doctor's snow vest.
del: remove most of the gear those jobs (nurse, doctor, research)spawn with and put it in their respective vendor.
code: created a parent for medical preset to avoid to having three time the same thing(back-end).

/:cl:

### CODE REVIEW

**Comment by cm13live** [comment]
This pull request has been mentioned on **CM-SS13**. Please provide feedback here:

https://forum.cm-ss13.com/t/adding-an-option-to-the-doctor-job-surgeon-feedback-on-my-pr-to-make-it-better/1206/1

File: code/game/jobs/job/civilians/support/doctor.dm [hunk 1]
```diff
@@ -1,3 +1,9 @@
+//job options for doctors surgeon pharmacy technician(preparation of medecine and distribution)
+
+#define DOCTOR_VARIANT "Doc"
+#define SURGEON_VARIANT "Sur"
+//#define PHARMACY_TECHNICIAN_VARIANT "Pha"
```
Comment (harryob) [inline code/game/jobs/job/civilians/support/doctor.dm:5]: no commented out code

File: code/game/jobs/job/civilians/support/doctor.dm [hunk 2]
```diff
@@ -8,8 +14,27 @@
...
+		doctor = TRUE
+		gear_preset = /datum/equipment_preset/uscm_ship/uscm_medical/doctor
+	else
+		doctor = FALSE
+		gear_preset = /datum/equipment_preset/uscm_ship/uscm_medical/doctor/surgeon
+
+//check what job option you took and generate the corresponding the good texte.
+/datum/job/civilian/doctor/generate_entry_message(mob/living/carbon/human/H)
+	if(doctor)
+		. = {"You're a commissioned officer of the USCM, though you are not in the ship's chain of command. <a href='"+URL_WIKI_DOC_GUIDE+"'>You are a doctor and tasked with keeping the marines healthy and strong, usually in the form of surgery.</a> You are a jack of all trade in all the medicals skills you can medicate and perform surgery on your patients. Your also know how to produce medecine If you do not know what you are doing, mentorhelp so a mentor can assist you."}
```
Comment (harryob) [inline code/game/jobs/job/civilians/support/doctor.dm:35]:
```suggestion
		. = {"You're a commissioned officer of the USCM, though you are not in the ship's chain of command. <a href='"+URL_WIKI_DOC_GUIDE+"'>You are a doctor and tasked with keeping the marines healthy and strong, usually in the form of surgery.</a> You are a jack of all trades in medicine: you can medicate, perform surgery and produce pharmaceuticals. If you do not know what you are doing, mentorhelp so a mentor can assist you."}
```

File: code/game/jobs/job/civilians/support/doctor.dm [hunk 3]
```diff
@@ -8,8 +14,27 @@
...
+	else
+		. = {"You're a commissioned officer of the USCM, though you are not in the ship's chain of command. <a href='"+URL_WIKI_DOC_GUIDE+"'>You are a surgeon and tasked with keeping the marines healthy and strong, usually in the form of surgery.</a> You are a doctor that specialize in surgery but you are also very capable when as a doctor. If you do not know what you are doing, mentorhelp so a mentor can assist you."}
```
Comment (harryob) [inline code/game/jobs/job/civilians/support/doctor.dm:37]:
```suggestion
		. = {"You're a commissioned officer of the USCM, though you are not in the ship's chain of command. <a href='"+URL_WIKI_DOC_GUIDE+"'>You are a surgeon and tasked with keeping the marines healthy and strong, usually in the form of surgery.</a> You are a doctor that  specializes in surgery, but you are also very capable in pharmacy and triage. If you do not know what you are doing, mentorhelp so a mentor can assist you."}
```

**Review by harryob — CHANGES_REQUESTED** [review]
@cmss13-devs/gameplay-architects  for the vendor changes

File: code/game/jobs/job/civilians/support/doctor.dm (hunk 1 above)
Comment (Huffie56) [reply code/game/jobs/job/civilians/support/doctor.dm:5]: a left over when i tried to make three possibility.

File: code/modules/gear_presets/uscm_medical.dm [hunk 4]
```diff
@@ -78,53 +86,31 @@
...
-	dress_gloves = list(/obj/item/clothing/gloves/marine/dress)
-	dress_shoes = list(/obj/item/clothing/shoes/laceup)
-
 /datum/equipment_preset/uscm_ship/uscm_medical/doctor/load_gear(mob/living/carbon/human/H)
 	var/backItem = /obj/item/storage/backpack/marine/satchel
 	if (H.client && H.client.prefs && (H.client.prefs.backbag == 1))
 		backItem = /obj/item/storage/backpack/marine
 
-	H.equip_to_slot_or_del(new /obj/item/device/radio/headset/almayer/doc(H), WEAR_L_EAR)
+	H.equip_to_slot_or_del(new backItem(H), WEAR_BACK)
```
Comment (morrowwolf) [inline code/modules/gear_presets/uscm_medical.dm:94]: snake case back_item

**Review by morrowwolf — CHANGES_REQUESTED** [review]
Needs to spawn with appropriate headset.

/obj/item/clothing/under/rank/medical needs removed

Remove reagent scanner HUD.

File: code/modules/gear_presets/uscm_medical.dm (hunk 4 above)
Comment (Huffie56) [reply code/modules/gear_presets/uscm_medical.dm:94]: > Needs to spawn with appropriate headset.
> 
> /obj/item/clothing/under/rank/medical needs removed
> 
> Remove reagent scanner HUD.

1-for the headset i double check and all the medical can get their headset from their vendor.
do you want me to reverse that and make them spawn with their headset anyway?
all headset are the same for nurse research and doc./obj/item/device/radio/headset/almayer/doc
 
2- unsure what you mean by /obj/item/clothing/under/rank/medical need to be removed as a lot of items are called like that.

3-do you want me to remove the reagent scanner HUD from all the vendors?

**Comment by Huffie56** [comment]
> Needs to spawn with appropriate headset.
> 
> /obj/item/clothing/under/rank/medical needs removed
> 
> Remove reagent scanner HUD.

1-for the headset i double check and all the medical can get their headset from their vendor.
do you want me to reverse that and make them spawn with their headset anyway?
all headset are the same for nurse research and doc./obj/item/device/radio/headset/almayer/doc
 
2- unsure what you mean by /obj/item/clothing/under/rank/medical need to be removed as a lot of items are called like that.

3-do you want me to remove the reagent scanner HUD from all the vendors?

File: code/modules/gear_presets/uscm_medical.dm (hunk 4 above)
Comment (Huffie56) [reply code/modules/gear_presets/uscm_medical.dm:94]: what you want me to do with H.equip_to_slot_or_del(new backItem(H), WEAR_BACK)?
do you want me to delete it? i honestly don't see what this line is doing....
look like some sort of check to put items in the backpack...

**Comment by morrowwolf** [comment]
So yes they should spawn with the headsets they should have.

The base item "/obj/item/clothing/under/rank/medical" should not be in any of their vendors. The subtypes are fine.

Reagent scanner HUD cannot be used by doctors so remove it from being vended from their vendors.

In our codebase we use snake case. So instead of "backItem" it should be named "back_item".

File: code/modules/gear_presets/uscm_medical.dm (hunk 4 above)
Comment (Huffie56) [reply code/modules/gear_presets/uscm_medical.dm]: it as been done

**Review by morrowwolf — APPROVED** [review]

**Pull request merged**